### ISSUE TITLE
PowerShell cmdlets need full documentation

### ISSUE THREAD

**Issue by jdhitsolutions** [microsoft/winget-cli#3822]
### Brief description of your issue

I'm sure this is on the to-do list, but the PowerShell commands need complete documentation. This includes descriptions, parameter information, and examples. Or is the documentation available as open-source repo where the community could help?

**Comment by denelon** [comment]
We will probably keep the documentation here at GitHub until the PowerShell modules are announced as GA (Generally Available). Once they are GA, we will add a section to Microsoft Learn. Both locations are open-source.

**Comment by jdhitsolutions** [comment]
That doesn't really help. As far as I can tell, the module has no intermediate markdown documents or external help files like you would get with the Platyps module. The help we get now is only what PowerShell automatically generates from the syntax and code. To be clear, I'm talking about command documentation for the user.

**Comment by jantari** [comment]
Yes, at the very least we need the `SYNOPSIS` for each cmdlet and `PARAMETER` descriptions for every parameter. These are what I reference the most at least. For a PowerShell function or a module written in PowerShell, this is easily done with [comment-based help](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comment_based_help?view=powershell-7.3#syntax-for-comment-based-help-in-script-modules), e.g. just paste this into PowerShell:

```powershell
function Find-WinGetPackage2 {
  <#
    .SYNOPSIS
    This cmdlet does something or other

    .PARAMETER Command
    Specifies the command to <whatever>

    .PARAMETER Count
    Specifies the count of something something
  #>
  [CmdletBinding()]
  Param (
    [string]$Command,
    [Uint32]$Count
  )

  Write-Output $Command
  Write-Output $Count
}

Get-Help Find-WinGetPackage2 -Detailed
```

Unfortunately I have never published a binary (C#) module but I know it is possible to add the same help-metadata to those. [These examples](https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/getprocesssample03-sample?view=powershell-7.3#example) use triple-slash `/// <summary>` comments, maybe that's all it takes?

**Comment by o-l-a-v** [comment]
PSResourceGet is a C# module with help available with `Get-Help` (might have to do `Update-Help -Module 'Microsoft.PowerShell.PSResourceGet'` first):

* <https://github.com/PowerShell/PSResourceGet>

<hr>

platyPS is a module that converts help written in Markdown to XML that could be used here:

* <https://github.com/PowerShell/platyPS>

As mentoined is PS Community Call 2024-02-15, platyPS is under heavy development:

* Note: <https://github.com/PowerShell/PowerShell-RFC/blob/master/CommunityCall/notes/20240215_Notes.md>
* Video with timestamp: <https://www.youtube.com/watch?v=OJlJJnCtFrQ&t=1670s>

Seems PSResourceGet uses platyPS:

* <https://github.com/search?q=repo%3APowerShell%2FPSResourceGet%20platyps&type=code>

<hr>

PSResourceGet also updates to learn.microsoft.com corresponding doc for a given cmdlet if for instance doing `Get-Help -Name 'Get-PSResource' -Online`. Winget does not have this yet, so `Get-Help -Name 'Assert-WinGetPackageManager' -Online` gives `Get-Help: The online version of this Help topic cannot be displayed because the Internet address (URI) of the Help topic is not specified in the command code or in the help file for the command.`.

**Comment by ThomasNieto** [comment]
@denelon there is a [PR](https://github.com/PowerShell/PowerShell/pull/23903) to include Microsoft.WinGet.CommandNotFound in PowerShell. That module requires this one so it really needs PowerShell based help and ideally updateable via #3206.

**Comment by denelon** [comment]
I met with some of our PowerShell folks to learn more about how we should be documenting our modules. We will be using platyPS. It's going to take some time to build this out and get it added to the pipeline. I'll add links to discrete Issues for each module to the top of this Issue as the primary one to track the request. We'll also be onboarding to Microsoft Learn so the documentation can be updated there as well.

**Comment by denelon** [comment]
I took a first pass at the platyPS documentation. I'd like to get it merged in so the engineering team can work on the build pipeline steps to get it included in the next build when we publish the modules.

I found several bugs and filed issues for them. Feel free to review the PR and tell me what I did wrong, but I wanted to go ahead and get the .md files in so others could contribute to improve them.

I'm expecting I'll also need to update a README somewhere with links to the content for platyPS and MAMAL over at Microsoft Learn.

**Comment by denelon** [comment]
We're cutting a new build with the help files.

**Comment by denelon** [comment]
Microsoft.WinGet.Client with new help is out:
https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/1.8.1911

We need to tackle Microsoft.WinGet.Configuration and Microsoft.WinGet.DSC next.

**Comment by denelon** [comment]
The [help files](https://github.com/microsoft/winget-cli/tree/master/src/PowerShell/Help) are now included in the build.

We still have plenty of work to improve the cmdlets and close gaps between the WinGet CLI, but I'm hoping this "helps" make them more understandable.
